### PR TITLE
Feature/169318693/top contributors widget (UI only - Storybook)

### PR DESCRIPTION
**Pull request by i-kushlianski**
UI part for Top Contributors - https://www.pivotaltracker.com/story/show/169318693.

Pls see only my commits at the end.

To be merged after @andrewchuhlomin's Welcome Widget.

### CODE REVIEW

File: packages/bootstrap/src/components/dashboard/widgets/TopContributors/styles.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
+import styled from 'styled-components';
+import { Button } from '../../../inputs';
+
+export const Header = styled.div`
+  height: 35px;
```
Comment (alex-melnik) [inline packages/bootstrap/src/components/dashboard/widgets/TopContributors/styles.tsx]: I'm not sure that we have to hardcode height of widget's header

File: packages/bootstrap/src/components/dashboard/widgets/TopContributors/TopContributors.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+} from './styles';
+
+interface ContributorProps {
+  userId?: string;
+  ranking?: number;
+  count?: number;
+  graphUrl?: string;
+}
+
+interface Columns {
```
Comment (vteremasov) [inline packages/bootstrap/src/components/dashboard/widgets/TopContributors/TopContributors.tsx]: Looks like it should be `type Columns = ContributorProps[][]`

File: packages/bootstrap/src/components/dashboard/widgets/TopContributors/styles.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+`;
+
+export const WidgetTitle = styled.span`
+  font-size: 22px;
+  font-weight: bold;
+  line-height: 35px;
+  color: #393941;
+`;
+
+// extract it to a separate component to be used by multiple widgets?
```
Comment (andrewchuhlomin) [inline packages/bootstrap/src/components/dashboard/widgets/TopContributors/styles.tsx]: I hope no, we don't need to do it

File: packages/bootstrap/src/components/dashboard/widgets/TopContributors/TopContributors.tsx [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+  );
+};
+
+const columns = 2;
+const totalCount = Object.keys(topContributors).length;
+const maxPerColumn = Math.ceil(totalCount / columns);
+
+let columnIndex = 1;
+
+const contributorsColumns = Object.entries(topContributors).reduce(
```
Comment (vteremasov) [inline packages/bootstrap/src/components/dashboard/widgets/TopContributors/TopContributors.tsx]: Looks like overcomplicated part

File: packages/bootstrap/src/components/dashboard/widgets/TopContributors/TopContributors.tsx [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+      </ContributorInfo>
+    </ContributorContainer>
+  );
+};
+
+const columns = 2;
+const totalCount = Object.keys(topContributors).length;
+const maxPerColumn = Math.ceil(totalCount / columns);
+
+let columnIndex = 1;
```
Comment (andrewchuhlomin) [inline packages/bootstrap/src/components/dashboard/widgets/TopContributors/TopContributors.tsx]: I reckon usually we start an index from 0

File: packages/bootstrap/src/components/index.spec.tsx [hunk 6]
```diff
@@ -20,35 +22,58 @@ const provider: WidgetProvider = {
...
+    });
+
+    expect((wrapper as any).toJSON()).toMatchSnapshot();
   });
 
-  it(`render Welcome`, async () => {
-    const widgetConfig = smeConfig.widgets['Welcome'] as WidgetConfig;
+  it(`renders TopContributors`, async () => {
+    const mockHistory = jest.spyOn(client, 'streamts');
+    const mockLucene = jest.spyOn(client, 'lucene');
```
Comment (andrewchuhlomin) [inline packages/bootstrap/src/components/index.spec.tsx]: Will be more convenient for you to use createClient?

File: packages/bootstrap/stories/widgets/TopContributorsContext.tsx [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    {
+      endpoint: '',
+      token: '',
+    },
+    {
+      forceHTTP: true,
+    },
+  );
+
+  client.lucene = async () => [1] as any;
```
Comment (andrewchuhlomin) [inline packages/bootstrap/stories/widgets/TopContributorsContext.tsx]:
```  const client = new Client(
    {
      endpoint: '',
      token: '',
    },
    {
      forceHTTP: true,
    },
  );

  client.lucene = async () => [1] as any;```

It is an unnecessary code block

File: packages/bootstrap/src/components/index.spec.tsx (hunk 6 above)
Comment (i-kushlianski) [reply packages/bootstrap/src/components/index.spec.tsx]: will do

File: packages/bootstrap/stories/widgets/TopContributorsContext.tsx (hunk 7 above)
Comment (i-kushlianski) [reply packages/bootstrap/stories/widgets/TopContributorsContext.tsx]: will remove

**Pull request merged**